src/Logging/Sink.php

```diff
@@ -0,0 +1,201 @@
+     *
+     * Example:
+     * ```
+     * $sink->exists();
+     * ```
+     *
+     * @param array $options Configuration Options
+     * @return bool
+     */
+    public function exists(array $options = [])
+    {
+        try {
+            $this->info($options);
+        } catch (NotFoundException $ex) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Delete the sink.
+     *
+     * Example:
+     * ```
+     * $sink->delete();
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/projects.sinks/delete projects.sinks delete API documentation.
+     * @codingStandardsIgnoreEnd
+     *
+     * @param array $options Configuration options.
+     */
+    public function delete(array $options = [])
+    {
+        $this->connection->deleteSink($options + [
+            'sinkName' => $this->formattedName
+        ]);
+    }
+
+    /**
+     * Update the sink. Please note this will trigger a network request if
+     * cached data is not available to perform the update with.
+     *
+     * Example:
+     * ```
+     * $sink->update([
+     *     'destination' => 'storage.googleapis.com/my-bucket'
+     * ]);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/projects.sinks/update projects.sinks update API documentation.
+     * @codingStandardsIgnoreEnd
+     *
+     * @param array $metadata {
+     *     The data to update.
+     *
+     *     @type string $destination The export destination. Please see
+     *           [Exporting Logs With Sinks](https://cloud.google.com/logging/docs/api/tasks/exporting-logs#about_sinks)
+     *           for more information and examples.
+     *     @type string $filter An [advanced logs filter](https://cloud.google.com/logging/docs/view/advanced_filters).
+     *     @type string $outputVersionFormat The log entry version to use for
+     *           this sink's exported log entries. This version does not have
+     *           to correspond to the version of the log entry when it was
+     *           written to Stackdriver Logging.
+     * }
+     * @param array $options Configuration options.
+     * @return array
+     */
+    public function update(array $metadata, array $options = [])
+    {
+        $options += $metadata;
+        $options += $this->info($options);
+
+        return $this->info = $this->connection->updateSink($options + [
+            'sinkName' => $this->formattedName
+        ]);
+    }
+
+    /**
+     * Retrieves the sink's details. If no sink data is cached a network
+     * request will be made to retrieve it.
+     *
+     * Example:
+     * ```
+     * $info = $sink->info();
+     * echo $info['destination'];
+     * ```
+     *
+     * @param array $options Configuration options.
+     * @return array
+     */
+    public function info(array $options = [])
```

Comment:
A link to https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/projects.sinks perhaps?